astropy/table/operations.py

```diff
@@ -1168,7 +1167,7 @@ def _join(left, right, keys=None, join_type='inner',
 
             out[out_name] = col_cls.info.new_like(cols, n_out, metadata_conflicts, out_name)
 
-            if not NUMPY_LT_1_17 or issubclass(col_cls, (Column, Time)):
```

Comment:
`not NUMPY_LT_1_17` will now always be true, so I think we can de-indent what follows and remove the `else` clause! (I always wondered why we didn't use `NUMPY_GE_<something>`!)

Reply:
Good catch!